app/models/content_consistency_checker.rb

```diff
@@ -0,0 +1,111 @@
+require 'gds_api/content_store'
+
+class ContentConsistencyChecker
+  attr_reader :content_id, :locale
+
+  def initialize(content_id, locale = "en")
+    @content_id = content_id
+    @locale = locale
+    @errors = []
+  end
+
+  def call
+    return @errors unless document
+
+    check_edition(draft, draft_content_store) if draft
+    check_edition(live, live_content_store) if live
+
+    @errors
+  end
+
+private
+
+  def item_from_content_store(path, content_store)
+    begin
+      content_store.content_item(path).parsed_content
+    rescue GdsApi::ContentStore::ItemNotFound, GdsApi::HTTPGone, GdsApi::HTTPForbidden
+      nil
+    end
+  end
+
+  def check_edition_existence(edition, content_item)
+    if edition.gone? && content_item
+      @errors << "Content exists in the content store."
+    elsif content_item.nil?
+      @errors << "Content is missing from the content store."
+    end
+  end
+
+  def check_edition(edition, content_store)
```

Comment:
So I think here we're not actually checking consistency correctly. For each `base_path` the Publishing API has there should be the equivalent at a `base_path` in the Content Store. Within these these the routes and redirects should match - well everything should match. So you could use a downstream presenter to compare each aspect of these.